src/components/app.tsx

```diff
@@ -403,15 +405,16 @@ export const App = () => {
     notifyStateIsDirty();
   };
 
-  const handleReset = () => {
+  const handleReset = useCallback(() => {
     if (confirm("Are you sure you want to reset?\n\nAny changes you have made will be lost.")) {
-      setGraph({nodes: [], edges: []});
+      setGraph(initialGraph ? {...initialGraph} : {nodes: [], edges: []});
     }
-  };
+  }, [initialGraph, setGraph]);
 
   const handleReturnToMainMenu = () => {
     if (confirm("Are you sure you want to go back to the main menu?\n\nAny changes you have made will be lost.")) {
       setGraph({nodes: [], edges: []});
+      setInitialGraph(undefined);
```

Comment:
I'm assuming Jie wants initial graph erased when going back to main menu instead of a saved state?

Reply:
Yep.